### PR TITLE
boards/pic32-clicker: cleanup uart configuration and initialization, cleanup header includes

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR cleans up the UART initialization and configuration similar to what was done in #12256.

This is an alternative to #12744.

This PR is untested. but maybe @francois-berder could give it a try.

### Testing procedure

- A green Murdock
- stdio UART works

### Issues/PRs references

Alternative to #12744 (and thus closes #12744), leftover from #12256

### CODE REVIEW

**Comment by francois-berder** [comment]
Thanks @aabadie for this PR.

I tested on a PIC32 clicker. You need to set ```STDIO_UART_DEV``` otherwise STDIO is not going to use the right UART device.
Here is the changes I made to make it work:
```diff --git a/boards/pic32-clicker/include/board.h b/boards/pic32-clicker/include/board.h
index c790e2560..9af476f7c 100644
--- a/boards/pic32-clicker/include/board.h
+++ b/boards/pic32-clicker/include/board.h
@@ -77,6 +77,13 @@ extern "C" {
  */
 void board_init(void);
 
+/**
+ * @brief   Use the 3rd UART for STDIO on this board
+ *
+ * This is the UART connected to the mikroBus.
+ */
+#define STDIO_UART_DEV      UART_DEV(3)
+
 #ifdef __cplusplus
 }
 #endif
```

File: boards/pic32-clicker/include/board.h [hunk 1]
```diff
@@ -40,6 +41,13 @@ extern "C" {
  */
 #define TICKS_PER_US (48)
 
+/**
+ * @brief   Use the 3rd UART for STDIO on this board
+ *
+ * This is the UART connected to the FTDI USB <-> UART device.
```
Comment (francois-berder) [inline boards/pic32-clicker/include/board.h]: This is the UART connected to the MikroBus.

**Comment by aabadie** [comment]
Thanks for having a look and testing quickly @francois-berder. I updated the PR with your comment and directly squashed.
Once the CI is green, we are good to go.

**Comment by aabadie** [comment]
Thanks, that was fast :)

**Pull request merged**